### PR TITLE
Add FXIOS-9841 Show search action in navigation toolbar on private homepage

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9841)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21604)

## :bulb: Description
- When the toolbar refactor is enabled, always show the search action in the navigation toolbar on private browsing mode homepage

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift [hunk 1]
```diff
@@ -688,15 +688,13 @@ final class ToolbarMiddleware: FeatureFlaggable {
...
-        guard !shouldShowDataClearanceAction else {
-            return dataClearanceAction
-        }
-
+        let canShowDataClearanceAction = canShowDataClearanceAction(isPrivate: isPrivateMode)
         let isNewTabEnabled = featureFlags.isFeatureEnabled(.toolbarOneTapNewTab, checking: .buildOnly)
-        let middleActionDefault = isNewTabEnabled ? newTabAction : homeAction
-        let middleActionHome = searchAction
-        let middleAction = url == nil ? middleActionHome : middleActionDefault
+        let isToolbarRefactorEnabled = featureFlags.isFeatureEnabled(.toolbarRefactor, checking: .buildOnly)
```
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift:693]: The check for toolbar refactor feature flag is not necessary: we are in the toolbar middleware which is only used when the refactor is on.

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift [hunk 2]
```diff
@@ -688,15 +688,13 @@ final class ToolbarMiddleware: FeatureFlaggable {
...
+        let middleActionForWebpage = canShowDataClearanceAction ?
+                                     dataClearanceAction : isNewTabEnabled ? newTabAction : homeAction
+        let middleActionForHomepage = isToolbarRefactorEnabled || !isPrivateMode ? searchAction : dataClearanceAction
+        let middleAction = url == nil ? middleActionForHomepage : middleActionForWebpage
```
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift:697]: The homepage should always show the search button - no matter if felt privacy is enabled or we are in private mode or one tap new tab is enabled or not.
```suggestion

        let middleActionForWebpage = canShowDataClearanceAction ?
                                     dataClearanceAction : isNewTabEnabled ? newTabAction : homeAction
        let middleActionForHomepage = searchAction
        let middleAction = url == nil ? middleActionForHomepage : middleActionForWebpage
```

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift (hunk 1 above)
Comment (MattLichtenstein) [reply firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift:693]: I knew that... 😅

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift (hunk 2 above)
Comment (MattLichtenstein) [reply firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift:697]: Yep, this was me taking the old toolbar into account 🤦‍♂️

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**